src/main/java/frc/robot/Constants.java

```diff
@@ -59,13 +62,27 @@ public class Constants {
                     MAX_ANGULAR_VELOCITY.in(Units.RotationsPerSecond),
                     MAX_ANGULAR_ACCELERATION.in(Units.RotationsPerSecond.per(Units.Second)));
 
-    public static final Pose2d[] OPTIMAL_POINTS_SHOOT = {
-        new Pose2d(1.97, 7.16, Rotation2d.fromDegrees(-161.57)),
-        new Pose2d(2.54, 3.06, Rotation2d.fromDegrees(-157.15)),
-        new Pose2d(4.16, 5.03, Rotation2d.fromDegrees(179.94))
-    };
-    public static final Pose2d[] OPTIMAL_POINTS_TRAP = {};
-    public static final Pose2d AMP_POSE = new Pose2d(0.0, 0.0, new Rotation2d(0, 0));
+    public static final List<Pose2d> OPTIMAL_POINTS_SHOOT_BLUE =
+            List.of(
+                    new Pose2d(1.97, 7.16, Rotation2d.fromDegrees(-161.57)),
+                    new Pose2d(2.54, 3.06, Rotation2d.fromDegrees(-157.15)),
+                    new Pose2d(4.16, 5.03, Rotation2d.fromDegrees(179.94)));
```

Comment:
Didn't we say you'll create the points from the JSON file? We'll have more than three points.

Reply:
We talked about a json? I don't see any reason.

Reply:
I talked about that with Barak. We though it would be nice to choose the shooting area in the GUI.

Reply:
Seems unnecessary, @rakrakon agrees.

Reply:
That's just it, for the dtop we don't necessarily want dozens of points. I predict we'll have 6 or 7.

Reply:
How so? With a 30cm square, this grid has 134 points:
![image](https://github.com/Galaxia5987/robot-2024/assets/31829093/a2839001-3997-41c1-968e-091db7c19e9d)

Reply:
Yes, but we only need a certain outline of points the distance of this outline will probably be the max shooting distance of the robot. Also i think after a certain number of points the effectiveness of adding more points will go down significantly (just like the number of cameras on the robot)

Reply:
> we only need a certain outline of points the distance of this outline will probably be the max shooting distance of the robot

But that way, if the robot's distance from the Speaker is smaller than the maximal shooting distance (i.e. close to the Speaker), the robot will needlessly drive farther from the Speaker, no?

Reply:
there will be point close to the shooter also

Reply:
How many in total? Is the grid in the screenshot I posted too large?

Reply:
No no no no no no no no. We only need the dtop when we're far from the speaker.

Reply:
No need to shout 🙂
DTOPing when you're close to the Speaker is a no-op. You can decide not to DTOP when within shooting range, but your code doesn't do that now.

Reply:
We'll implement it when we do the interpolation map for the shooter.

Reply:
How are these related? Why not implement this now and check in simulation it works?

Reply:
10c14aa